shpkpr/deployment/bluegreen/deployment.py

```diff
@@ -143,7 +144,16 @@ def get_deployment_group(app_definition):
             return labels.get('HAPROXY_DEPLOYMENT_GROUP')
 
         target_deployment_group = get_deployment_group(self.app_definition)
-        for app_definition in self.marathon_client.list_applications():
+        for app_definition in _fetch_apps_state:
             if get_deployment_group(app_definition) == target_deployment_group:
                 return app_definition
         return None
+
+    fetch_app_states_memo = None
+    def _fetch_app_states(self):
```

Comment:
We already depend on https://pypi.python.org/pypi/cached-property in shpkpr, that might be better here